src/syncModule.ts

```diff
@@ -758,63 +758,83 @@ export class TodoistSync  {
         }
 
         // 将新事件合并到现有事件中并保存到 JSON
-        
+
         await this.plugin.cacheOperation.appendEventsToCache(processedEvents)
         this.plugin.saveSettings()
         } catch (error) {
         console.error('同步任务状态时出错：', error)
         }
     }
 
+    async syncNewTasksToObsidian(unSynchronizedEvents){
+        console.log("starting synchronize tasks from todoist to obsidian")
+        try {
+            const processedEvents = []
+            for (const e of unSynchronizedEvents) {
+                console.log(e)
+                await this.plugin.fileOperation?.syncNewTaskToTheFile(e)
+                new Notice(`Task ${e.object_id} is added.`)
+                processedEvents.push(e)
+            }
+
+            await this.plugin.cacheOperation.appendEventsToCache(processedEvents)
+            this.plugin.saveSettings()
+        } catch (error) {
+            console.error('同步任务状态时出错：', error)
+        }
+        console.log("synchronize tasks from todoist to obsidian finished")
+    }
+
 
     async syncTodoistToObsidian(){
         try{
             const all_activity_events = await this.plugin.todoistSyncAPI.getNonObsidianAllActivityEvents()
-            
+
             // remove synchonized events
             const savedEvents = await this.plugin.cacheOperation.loadEventsFromCache()
             const result1 = all_activity_events.filter(
             (objA) => !savedEvents.some((objB) => objB.id === objA.id)
             )
-    
-           
+
             const savedTasks = await this.plugin.cacheOperation.loadTasksFromCache()
             // 找出 task id 存在于 Obsidian 中的 task activity
             const result2 = result1.filter(
-            (objA) => savedTasks.some((objB) => objB.id === objA.object_id)
+            (objA) => !savedTasks.some((objB) => objB.id === objA.object_id)
             )
             // 找出 task id 存在于 Obsidian 中的 note activity
             const result3 = result1.filter(
-                (objA) => savedTasks.some((objB) => objB.id === objA.parent_item_id)
+                (objA) => !savedTasks.some((objB) => objB.id === objA.parent_item_id)
                 )
-        
-    
-    
-    
+
+            console.log(result1)
+
             const unsynchronized_item_completed_events = this.plugin.todoistSyncAPI.filterActivityEvents(result2, { event_type: 'completed', object_type: 'item' })
             const unsynchronized_item_uncompleted_events = this.plugin.todoistSyncAPI.filterActivityEvents(result2, { event_type: 'uncompleted', object_type: 'item' })
 
             //Items updated (only changes to content, description, due_date and responsible_uid)
             const unsynchronized_item_updated_events = this.plugin.todoistSyncAPI.filterActivityEvents(result2, { event_type: 'updated', object_type: 'item' })
+            const unsynchronized_item_added_events = this.plugin.todoistSyncAPI.filterActivityEvents(result2, { event_type: 'added', object_type: 'item' })
```

Comment:
Extract new tasks events that need to be synchronized to obsidian from the result4 variable.
```suggestion
            const unsynchronized_item_added_events = this.plugin.todoistSyncAPI.filterActivityEvents(result4, { event_type: 'added', object_type: 'item' })
```